plutus-book/doc/09-extended.adoc

```diff
@@ -177,114 +225,64 @@ together out of transaction and ledger data,
-    -> RedeemerScript
-    -- ^ the redeemer
-    -> m [String]
-----
-
-We will explain the purpose of the argument of type `DataScripts` in the note below.
-These are data scripts that are associated with script addresses the carrying
-transaction is paying _to_.
-Now, let us look more closely at how these scripts fit into the extended model
-transaction and UTxO structure.
 
-Recall that in the classic model, UTxO entries are of the form
-`(txid, ix) |-> (a,c)`. The pair `(a,c)` has type `TxOut` and is called an output,
-where `a` is the address of the owner of the funds
-(the paying or staking public key, usually), and `c` is the value of Ada in the output.
 
-For the extended model, we say that an output is locked by (or belongs to)
-a script if `a` is a script address (hash of the validator script). In this
-case, we must store additional data on the
-ledger in the `TxOut` - the data script. It must be stored in full, not just the
-hash, since it will be passed as an argument to the validator during
-validation of a transaction which is trying to spend this output.
-The extended `TxOut` for a script address contains
+Let us now consider how our example above conforms to this validator-data-redeemer
+script model. The parts of the contract we saw above, `s`, `hs`, and `wg`, are
+defined in `PlutusTx`, and are compiled into `Plutus Core` scripts. The
 
-- an address `a`, which is the hash of the validator
-- the value `c` of funds at this address
-- the type of the output, and the data script, expressed as `PayToScript datascript`:
+* The _validator_ script is `s`
+* The _data_ script is `hs`
+* The _redeemer_ script is `wg`
 
-A `TxOut` which belongs to a public key address contains only the address and the value.
-Thus, a term of the `TxOut` data structure type includes a term of the type:
+Then, the funds can be spent when
 
 ----
-data TxOutType =
-    PayToScript !DataScript -- A pay-to-script output with the given data script.
-    | PayToPubKey !PubKey -- A pay-to-pubkey output.
+s hs wg
 ----
 
-This is how the outputs are structured both inside the carrying transaction
-and the UTxO. In the extended model, the position of an output in the outputs list
-carried by a transaction is its reference index `ix`.
-In the UTxO, the outputs are stored as a finite
-map with the keys of the form `(txid, ix)`. Once a transaction is validated,
-the update mechanism
-of the UTxO is the same for both the the extended and classic model (with the
-extended model storing the additional `TxOutType` data described above).
-For a reminder, see the <<03-UTXO#03-UTXO>> chapter.
+evaluates to `True`. The `PendingTx` value is not used in this example, but
+we could have extended the contract logic to only allow a transaction to
+spend the funds under the additional condition that it is doing so after
+after a certain slot. This `currentSlot` data would be passed via `PendingTx`.
```

Comment:
You need to say a little more for this to be comprehensible to someone who doesn't know what's in `PendingTx`. Plus this makes it sound as though `PendingTx` contains the current slot, which it doesn't.